deepchecks/tabular/checks/data_integrity/feature_feature_correlation.py

```diff
@@ -0,0 +1,102 @@
+from deepchecks.utils.dataframes import generalized_corrwith
+
+__all__ = ['FeatureFeatureCorrelation']
+
+
+class FeatureFeatureCorrelation(SingleDatasetCheck):
+    """Checks for pairwise correlation between the features.
+
+        Extremely correlated pairs could indicate redundancy and even duplication.
+
+    Parameters
+    ----------
+        n_top_pairs : int , default: 5
+        Number of pairs to show, sorted by the correlation strength
+    """
+
+    def __init__(self,
+                 n_top_pairs: int = 5,
+                 **kwargs):
+        super().__init__()
+        self.n_top_pairs = n_top_pairs
+
+    def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
+        """
+        Run Check.
+
+        Returns
+        -------
+        CheckResult
+        A heatmap of the pairwise correlations between the features
+        """
+        if dataset_type == 'train':
+            dataset = context.train
+        else:
+            dataset = context.test
+
+        dataset.assert_features()
+        num_features = dataset.numerical_features
+        cat_features = dataset.cat_features
+
+        encoded_cat_data = dataset.data.loc[:, cat_features].apply(lambda x: pd.factorize(x)[0])
+
+        # Numerical-numerical correlations
+        num_num_corr = dataset.data.loc[:, num_features].corr(method='spearman')
+
+        # Numerical-categorical correlations
+        num_cat_corr = generalized_corrwith(dataset.data.loc[:, num_features], encoded_cat_data,
+                                            method=correlation_ratio)
+
+        # Categorical-categorical correlations
+        cat_cat_corr = encoded_cat_data.corr(method=symmetric_theil_u_correlation)
+
+        # Compose results from all the features
+        all_features = num_features + cat_features
+        full_df = pd.DataFrame(index=all_features, columns=all_features)
+        full_df.loc[num_features, num_features] = num_num_corr
+        full_df.loc[cat_features, cat_features] = cat_cat_corr
+        full_df.loc[num_features, cat_features] = num_cat_corr
+        full_df.loc[cat_features, num_features] = num_cat_corr.transpose()
+
+        # Display
+        fig = px.imshow(full_df)
+
+        return CheckResult(value=full_df, header='Feature-Feature Correlation', display=fig)
+
+    def add_condition_max_number_of_pairs_above(self, threshold: float = 0.9, n_pairs: int = 0):
+        """ Add condition that all pairwise correlations are less than threshold, except for the diagonal."""
+        def condition(result):
+            results_ge = result.ge(threshold)
+            high_corr_pairs = []
+            for i in results_ge.index:
+                for j in results_ge.columns:
+                    if (i == j) | ((j, i) in high_corr_pairs):
+                        continue
+                    if results_ge.loc[i, j]:
+                        high_corr_pairs.append((i, j))
```

Comment:
Feels like this could be done more efficiently, or at least cleaner, than a double loop. For the least, can use apply and get series name, I think, but maybe there's a better way to get pairs of (index, column) from data frame (and then mask to take only from True, and later filter out duplicates)

Reply:
I looked for a better way but couldn't find any that was still legible - open for suggestions

Reply:
Try these suggestions maybe?
https://stackoverflow.com/questions/26854091/getting-index-column-pairs-for-true-elements-of-a-boolean-dataframe-in-pandas